lib/src/facts/posix/networking_resolver.cc

```diff
@@ -106,7 +110,7 @@ namespace facter { namespace facts { namespace posix {
                     result.domain.assign(parts[1].begin(), parts[1].end());
                     return false;
                 }
-                if (parts[0] == boost::as_literal("search")) {
+                if (search.empty() && parts[0] == boost::as_literal("search")) {
```

Comment:
Noticed a discrepancy between the comment and the code: it was respecting the last `search` directive and not the first.  Can split into a separate commit if desired.